packages/grid/x-data-grid/src/hooks/features/filter/useGridFilter.tsx

```diff
@@ -386,38 +390,48 @@ export const useGridFilter = (
 
   const flatFilteringMethod = React.useCallback<GridStrategyProcessor<'filtering'>>(
     (params) => {
-      if (props.filterMode === 'client' && params.isRowMatchingFilters) {
-        const tree = gridRowTreeSelector(apiRef);
-        const rowIds = (tree[GRID_ROOT_GROUP_ID] as GridGroupNode).children;
-        const filteredRowsLookup: Record<GridRowId, boolean> = {};
-        const filterCache = {};
-
-        for (let i = 0; i < rowIds.length; i += 1) {
-          const rowId = rowIds[i];
-          let isRowPassing;
-          if (typeof rowId === 'string' && rowId.startsWith('auto-generated-group-footer')) {
-            isRowPassing = true;
-          } else {
-            const { passingFilterItems, passingQuickFilterValues } =
-              params.isRowMatchingFilters(rowId);
-            isRowPassing = passFilterLogic(
-              [passingFilterItems],
-              [passingQuickFilterValues],
-              params.filterModel,
-              apiRef,
-              filterCache,
-            );
-          }
-          filteredRowsLookup[rowId] = isRowPassing;
-        }
+      if (props.filterMode !== 'client' || !params.isRowMatchingFilters) {
         return {
-          filteredRowsLookup,
+          filteredRowsLookup: {},
           filteredDescendantCountLookup: {},
         };
       }
 
+      const rows = apiRef.current.caches.rows.rows;
```

Comment:
Can we avoid adding `caches.rows.rows`?
If the goal was to avoid calling `Object.values(apiRef.current.state.rows.dataRowIdToModelLookup)` in `flatFilteringMethod` for performance reasons, maybe do it in a selector outside of the `flatFilteringMethod`?